### PR TITLE
Added a feature for anonymous commenting

### PR DESCRIPTION
**Pull request by sewagodimo**
The feature for anonymous commenting was originally hard coded but now the admin can decide what he alias given to anonymous commenter is.

### CODE REVIEW

File: molo/core/models.py [hunk 1]
```diff
@@ -1441,7 +1441,10 @@ class ArticlePage(ImportableMixin, CommentedPageMixin,
         null=True)
     commenting_open_time = models.DateTimeField(null=True, blank=True)
     commenting_close_time = models.DateTimeField(null=True, blank=True)
-
+    commenting_anonymous_alias = models.CharField(
+        max_length=100,
+        blank=True,
```
Comment (codiebeulaine) [inline molo/core/models.py]: you don't need blank to be true is there is a default

File: molo/core/models.py [hunk 2]
```diff
@@ -1532,6 +1535,12 @@ def is_commenting_enabled(self):
             return False
         return True
 
+    def get_anonymous_commenting_alias(self):
```
Comment (codiebeulaine) [inline molo/core/models.py]: I don't think this method is needed if blank is false and the default is provided

**Comment by codiebeulaine** [comment]
@sewagodimo why is this on an ArticlePage level? Will the alias be different per article or will there be one alias per site?

File: molo/core/models.py [hunk 3]
```diff
@@ -1441,7 +1441,10 @@ class ArticlePage(ImportableMixin, CommentedPageMixin,
         null=True)
     commenting_open_time = models.DateTimeField(null=True, blank=True)
     commenting_close_time = models.DateTimeField(null=True, blank=True)
-
+    commenting_anonymous_alias = models.CharField(
+        max_length=100,
+        blank=True,
+        default='anonymous')
```
Comment (codiebeulaine) [inline molo/core/models.py]: what happens on other languages that are not english? What does anonymous evaluate to? Does this get translated int he templates with {% trans %} tag?

File: molo/core/tests/test_models.py [hunk 4]
```diff
@@ -345,6 +345,17 @@ def test_commenting_enabled(self):
             commenting_state=constants.COMMENTING_DISABLED)
         self.assertFalse(article_3.is_commenting_enabled())
 
+    def test_commenting_anonymous_alias(self):
```
Comment (codiebeulaine) [inline molo/core/tests/test_models.py]: can we also test that when making a comment, that the anonymous alias is used, and that it is the correct one being used. Can we also do a test for a translated version of anonymous?

File: molo/core/models.py (hunk 3 above)
Comment (sewagodimo) [reply molo/core/models.py]: in molo-gem this is translated. will the tag work here? I wasnt sure about that

File: molo/core/tests/test_models.py (hunk 4 above)
Comment (sewagodimo) [reply molo/core/tests/test_models.py]: yes I will addd that

File: molo/core/models.py (hunk 3 above)
Comment (codiebeulaine) [reply molo/core/models.py]: @sewagodimo unless the template is changed to use the commenting_anonymous_alias, the current functionality won't change. What you would need to do is add something like {% trans commenting_anonymous_alias %} if it is the default

Comment (codiebeulaine) [reply molo/core/models.py]: Happy to jump on a call sometime tomorrow and step through it

**Comment by sewagodimo** [comment]
I have closed this pull request and moved and made another one under molo.commenting

**Pull request closed without merging**